### PR TITLE
Implement search and differentiate PUT from PATCH #1, #4, #9

### PR DESCRIPTION
**Pull request by SpraylnlPray**
Implementation for a user to search for projects using a string, or just filter projects depending on their active status.

Implementation for an admin user to change the active status of a project.

Tests are provided, feedback is welcome.

### CODE REVIEW

File: bpm_projects_api/apis/project.py [hunk 1]
```diff
@@ -101,3 +155,35 @@ def delete(self, uid):
     def put(self, uid):
         """Update a project given its identifier"""
         return dao.update(uid, ns.payload)
+
+
+# route is chosen because of the example above the "T odo" class on
+# https://flask-restplus.readthedocs.io/en/0.11.0/example.html and
+# https://www.rithmschool.com/courses/flask-fundamentals/routing-with-flask
+# "<url>/<<data_type>:<variable_name>>
```
Comment (enriquezrene) [inline bpm_projects_api/apis/project.py]: do we need these comments?

**Comment by SpraylnlPray** [comment]
Please have in mind that the function for changing a projects' active property is not finished

File: bpm_projects_api/apis/project.py (hunk 1 above)
Comment (SpraylnlPray) [reply bpm_projects_api/apis/project.py]: Those were for me so that i find the source again, they can be removed

**Comment by EliuX** [comment]
There is too many responsibilities/code added to a couple of python modules. Try to structure move those well defined features to new modules in the same package.

**Comment by EliuX** [comment]
Reading `flake tests` it seems that you programmed some tests for flake8 as you wrote tests for pytest. I would recommend something like "Refactor to comply flake8 checks", which means that you changed your code based on all the checks flake8 did to your code.

File: bpm_projects_api/apis/dao.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        project = self.get(id)
+        self.projects.remove(project)
+
+    def search(self, search_criteria):
+        matching_projects = self.select_matching_projects(search_criteria)
+
+        if len(matching_projects) > 0:
+            return matching_projects
+        else:
+            return '', 404
```
Comment (EliuX) [inline bpm_projects_api/apis/dao.py]: The data layer should not be involved with the connection layer. This DAO is couple to rest results, even more with Flask RestPlus kind of results that returns a (result, http response code)

**Pull request merged**